### ISSUE TITLE
WebView data sharing between InfoActivity and BrowserFragment

### ISSUE THREAD

**Issue by ahunt** [mozilla-mobile/focus-android#199]
WebView only uses one cache per application. We're using WebView for both the main focus browser, and also for information screens (e.g. SUMO pages describing how to set a default browser, explaining telemetry, etc.).

We need to figure out what we can to do avoid issues with the same cache/cookies storage/etc being used for both

**Comment by ahunt** [comment]
Previous discussions:
- The patch landing InfoActivity:
https://github.com/mozilla-mobile/focus-android/pull/183#discussion_r104998888
- Discussion re: WebViewProvider for InfoActivity:
 in #182

**Comment by pocmo** [comment]
Talking with antlam about UI I just thought: Why aren't we just using custom tabs here? We can style them as we want and we wouldn't share the browsing data with our WebView. And soon Firefox will support custom tabs too. :)

(Of course this makes only sense for external URLs, like SUMO)

**Comment by ahunt** [comment]
No reason no to use custom tabs, I'm guessing every device will at least have chrome supporting that?

But: that means we'll have one style for internal pages (about/your rights), and a different one for the rest (Help, Telemetry->Learn More, About->Learn More, Your Rights->All links). I'm guessing that's not too bad?

**Comment by ahunt** [comment]
One issue with custom tabs is: what if the default browser doesn't support custom tabs? In that case we're essentially sending a VIEW intent, and dumping people into a different app.

We can manually figure out what CT supporting browsers are available, but that makes the code more complicated - and it's not guaranteed to work if there's no CT supporting browse (which hopefully shouldn't be the case since people will usually have an up to date chrome or firefox installed).

**Comment by ahunt** [comment]
It looks like code to solve that already exists:
https://github.com/GoogleChrome/custom-tabs-client/blob/c51efbddc4f976c88d15c730a79feba65ca857af/shared/src/main/java/org/chromium/customtabsclient/shared/CustomTabsHelper.java#L63

**Comment by ahunt** [comment]
This might be worth discussing while triaging? It looks like we won't use CT for now.

**Comment by pocmo** [comment]
Yeah, I was wondering whether we want to at least use the blocking code in InfoActivity. If we load a tracker in InfoActivity (let's say SUMO uses Google Analytics) - will we still block it during the next normal browse or is it loaded from cache without us being able to block it?

**Comment by ahunt** [comment]
I think we also need error page support for InfoActivity (mostly for no-network situations), so using the full WebViewProvider (or at least our FocusWebViewClient) would be nice.  (We can't really use FocusWebViewClient directly, since that depends on the webkit build, hence we should use the full WebViewProvider?)

**Comment by ahunt** [comment]
I've got a branch that replaces WebView with WebFragment. That has the benefit of using the same tracking protection settings, and it adds a progress bar too. I accidentally build that on top of my path from #250, so I'll wait until we've decided that before rebasing/dropping as needed (removing just that patch causes rebase conflicts, so I'll avoid dealing with that until we know what we're doing about the error page non-completion).
